src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
Can't users hold onto buffers in the pool anyways (after they've been "returned"), and read from them regardless? Or do we protect from that in the implementation somehow?